### PR TITLE
SDK V2

### PR DESCRIPTION
**Pull request by mt-max**
You can review the codebase by commits.

Feel free to checkout the branch directly to test it.

Since it is not published yet and you want to try it in your project, you can add the following line in your project dependency
```
"@moneytree/mt-link-javascript-sdk": "moneytree/mt-link-javascript-sdk#2.0.0-beta"
```

I hosted the documentation on my fork static github page, please read it and comment on any grammar, spelling mistakes, or rephrase if necessary (will setup one for production when this PR got merged)

Documentation: https://mt-max.github.io/mt-link-javascript-sdk/#/

@mt-aanwar, not clue why I can't tag you as the reviewer, so I tagged you here.

### CODE REVIEW

**Comment by mt-rory** [comment]
Tried to implement on Vault.
Few concerns here:
- `mtLinkSdk.init(clientId, options)` this clientId it means brand_client_id? or Vault’s app_id?
- Vault also pass `brand_client_id`, `config.sdk_platform` and `locale` to myAccount
   But I didn't see these parameters on the document.

**Comment by mt-max** [comment]
I have updated the codebase and doc based on @mt-rory feedback.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+<h6>Usage:</h6>
+
+```javascript
+mtLinkSdk.init(clientId, options);
+```
+
+| Parameter | Type | Required | Default Value | Description |
+| - | - | - | - | - |
+| clientId | string | true | | <p>OAuth `clientId` of the app (please request this from your Moneytree representative if you need one.<p><p><strong>NOTE:</strong> SDK will throw an error if non was provided.</p> |
```
Comment (mt-dfcowell) [inline docs/README.md:61]: Typo (no closing parenthesis.)

Comment (mt-max) [reply docs/README.md]: updated

**Comment by jackyLianMT** [comment]
@mt-max 
do we have a document for what kind of internal setup needed for setting up the Fast onboarding?
also https://mt-max.github.io/mt-link-javascript-sdk/#/?id=requestmagiclink
requestmagiclink also require configurations to work right, we have to specify that in the doc

**Comment by mt-rory** [comment]
> @mt-max
> do we have an internal document for what kind of setup needed for setting up the Fast onboarding with a client?
> also https://mt-max.github.io/mt-link-javascript-sdk/#/?id=requestmagiclink
> requestmagiclink also require configurations to work right? if that is the case, we have to specify that in the doc

Jacky for onBoadring API, need to setup Client TOS link on Active Admin
Screen Shot 2020-07-17 at 17 39 32

**Comment by mt-rory** [comment]
I did fully test, tried to implement on Vault. 
The document is pretty clear to me.
And There're two errors that may need to have a look.

- onboard API
  Got an error on console, but it's from MyAccount. But the API `https://myaccount-develop.getmoneytree.com/onboard`  it returns 200 success, so I assume it's work but something wrong on MyAccount.
   **Max addressed this fixed on MyAccount [PR](https://github.com/moneytree/mt-guest-service/pull/1349)**
Screen Shot 2020-07-17 at 18 03 04

- requestMagicLink API
 Got the status of 500 error with the following post request. 
 But I tried to change client_id with `1b374ffc7d08132517da3c3cf635889cf172ca2da6db3e8a01c87fa349fe88bf` Then it works.
```curl 'https://myaccount-develop.getmoneytree.com/magic-link.json?client_id=c88fb2a60f02ded818d436110b20afb1a956b7d3d6c9fbd7a0cef53dd4391694&configs=sdk_platform%3Djs%26sdk_version%3D2.0.0-beta%26back_to%3Dhttps%253A%252F%252Flocalhost%253A9000' \
  -H 'Connection: keep-alive' \
  -H 'Pragma: no-cache' \
  -H 'Cache-Control: no-cache' \
  -H 'sec-ch-ua: "\\Not\"A;Brand";v="99", "Chromium";v="84", "Google Chrome";v="84"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.68 Safari/537.36' \
  -H 'Content-Type: application/json' \
  -H 'Accept: */*' \
  -H 'Origin: https://localhost:9000' \
  -H 'Sec-Fetch-Site: cross-site' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Referer: https://localhost:9000/' \
  -H 'Accept-Language: en-US,en;q=0.9,zh-TW;q=0.8,zh;q=0.7' \
  --data-binary '{"email":"rlin@moneytree.jp","magic_link_to":"/settings"}' \
  --compressed```

**Comment by mt-max** [comment]
Only onboard requires special setup which was mentioned in the doc, as for magic link, since this is for web and magic link is just an url, so it works without the need for configuration.

**Comment by jackyLianMT** [comment]
@mt-max is this PR ready for merge?

**Comment by mt-aanwar** [comment]
Before merging this, do we know if the new web SDK introduce any breaking changes?

I'm asking this because the Integration team might need to announce to the clients (who uses our SDK) that a new version will be released and using `latest` version may break their implementation.

Below screenshot is taken from our current README:
Screen Shot 2020-07-21 at 9 12 43 am

**Comment by mt-max** [comment]
Yes, since this is version 2.0.0-beta, it will be a breaking change, but since I am tagging it beta, they have to expect some small bugs here and there (we have tested it, it is quite stable, but who knows if someone might triggered some edge cases).

**Comment by jackyLianMT** [comment]
Furhter information from Max:
`If we don't tag it then it will be latest, if we tag it to something else, e.g: beta or next, then latest will not point to it.`

**Comment by mt-dfcowell** [comment]
Ok, I think this is good to merge. I don't have permissions on this repo, otherwise I'd do it.

**Comment by jackyLianMT** [comment]
@mt-max feel free to merge provided the release is tagged as 2.0-beta or something similar.

@mt-aanwar there are changes required on the my account to make sure we are not accidentally releasing the fast onboarding to other clients. So let's do not release this to any client yet. We will have a release planning meeting at next week to discuss how to roll this out.

**Comment by mt-aanwar** [comment]
> Furhter information from Max:
> `If we don't tag it then it will be latest, if we tag it to something else, e.g: beta or next, then latest will not point to it.`

I like this idea. Let's tag it as either `beta` or `next` then :)

**Pull request merged**